pkg/policyviolation/namespacedpv.go

```diff
@@ -130,7 +129,7 @@ func (nspv *namespacedPV) updatePV(newPv, oldPv *kyverno.PolicyViolation) error
 	logger := nspv.log.WithValues("policy", newPv.Spec.Policy, "kind", newPv.Spec.ResourceSpec.Kind, "namespace", newPv.Spec.ResourceSpec.Namespace, "name", newPv.Spec.ResourceSpec.Name)
 	var err error
 	// check if there is any update
-	if reflect.DeepEqual(newPv.Spec, oldPv.Spec) {
+	if !hasViolationSpecChanged(newPv.Spec.DeepCopy(), oldPv.Spec.DeepCopy()) {
 		logger.V(4).Info("policy violation spec did not change, not upadating the resource")
```

Comment:
same as above - add relevant data to log

Reply:
same here, logger is configured.